src/models.rs

```diff
@@ -510,23 +533,64 @@ pub struct BatchMetadata {
     /// Batch statistics
     pub batch_stats: BatchStats,
     /// Current state of the batch
-    pub state: String,
+    pub state: BatchState,
     /// Name of the batch (duplicate)
     pub name: String,
+    /// The output configuration for the batch.
+    pub output: Option<OutputConfig>,
+}
+
+/// The state of a batch operation.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum BatchState {
+    BatchStatePending,
+    BatchStateInitializing,
+    BatchStateRunning,
+    BatchStateSucceeded,
+    BatchStateFailed,
+    BatchStateCancelled,
+    BatchStateExpired,
+    BatchStateUnspecified,
 }
 
 /// Statistics for the batch
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct BatchStats {
     /// Total number of requests in the batch
-    pub request_count: String,
+    #[serde(deserialize_with = "from_str_to_i64")]
+    pub request_count: i64,
     /// Number of pending requests
-    pub pending_request_count: Option<String>,
+    #[serde(default, deserialize_with = "from_str_to_i64_optional")]
+    pub pending_request_count: Option<i64>,
     /// Number of completed requests
-    pub completed_request_count: Option<String>,
+    #[serde(default, deserialize_with = "from_str_to_i64_optional")]
+    pub completed_request_count: Option<i64>,
     /// Number of failed requests
-    pub failed_request_count: Option<String>,
+    #[serde(default, deserialize_with = "from_str_to_i64_optional")]
+    pub failed_request_count: Option<i64>,
+    /// Number of successful requests
+    #[serde(default, deserialize_with = "from_str_to_i64_optional")]
+    pub successful_request_count: Option<i64>,
+}
+
+fn from_str_to_i64<'de, D>(deserializer: D) -> Result<i64, D::Error>
```

Comment:
This solution allows for much simpler code on the calling side and helps us follow a fail-fast strategy.